src/dialect.ml

```diff
@@ -0,0 +1,181 @@
+open! Stdune
+
+module File_kind = struct
+  type t =
+    { kind       : Ml_kind.t
+    ; extension  : string
+    ; preprocess : (Loc.t * Action_dune_lang.t) option
+    ; format     : (Loc.t * Action_dune_lang.t * string list) option
+    }
+
+  let to_dyn { kind ; extension ; preprocess ; format } =
+    let open Dyn.Encoder in
+    record
+      [ "kind"      , Ml_kind.to_dyn kind
+      ; "extension" , string extension
+      ; "preprocess", option (pair Loc.to_dyn Action_dune_lang.to_dyn) preprocess
+      ; "format"    , option (triple Loc.to_dyn Action_dune_lang.to_dyn (list string)) format
+      ]
+end
+
+type t =
+  { name       : string
+  ; file_kinds : File_kind.t Ml_kind.Dict.t
+  }
+
+let name t = t.name
+
+let to_dyn { name ; file_kinds } =
+  let open Dyn.Encoder in
+  record
+    [ "name"      , string name
+    ; "file_kinds", Ml_kind.Dict.to_dyn File_kind.to_dyn file_kinds
+    ]
+
+let decode =
+  let open Dune_lang.Decoder in
+  let kind kind =
+    let+ loc, extension  = field "extension" (located string)
+    and+ preprocess = field_o "preprocess" (located Action_dune_lang.decode)
+    and+ format     = field_o "format" (map ~f:(fun (loc, x) -> loc, x, []) (located Action_dune_lang.decode))
```

Comment:
Could you stick to an 80 char limit? It makes reading diffs much easier.

Reply:
Sure, it is fixed.